core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -260,6 +260,40 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
                 createClassLoader(paths, dependencyLoader, atts), disableFile, dependencies, optionalDependencies);
     }
 
+    private static void fix(Attributes atts, List<PluginWrapper.Dependency> optionalDependencies) {
+        // don't fix the dependency for yourself, or else we'll have a cycle
+        String pluginName = atts.getValue("Short-Name");
+        
+        // some earlier versions of maven-hpi-plugin apparently puts "null" as a literal in Hudson-Version. watch out for them.
+        String jenkinsVersion = atts.getValue("Jenkins-Version");
+        if (jenkinsVersion==null)
+            jenkinsVersion = atts.getValue("Hudson-Version");
+        
+        optionalDependencies.addAll(getImpliedDependencies(pluginName, jenkinsVersion));
+    }
+    
+    /**
+     * Returns all the plugin dependencies that are implicit based on a particular Jenkins version
+     */
```

Comment:
As a rule in PRs we just leave it as `TODO`, with the expectation that the person merging the PR should replace that with the actual version then in the POM. Of course in this case you expect the version to be 2.0 (or some uninteresting alpha variant thereof).